action/protocol/poll/staking_committee.go

```diff
@@ -74,26 +83,27 @@ func (sc *stakingCommittee) DelegatesByHeight(height uint64) (state.CandidateLis
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get native chain candidates")
 	}
-	// merge the candidates
-	if err := sc.mergeDelegates(cand, nativeVotes); err != nil {
-		return nil, errors.Wrap(err, "failed to merge candidates")
-	}
-	return cand, nil
+	return sc.mergeDelegates(cand, nativeVotes), nil
 }
 
 func (sc *stakingCommittee) ReadState(ctx context.Context, sm protocol.StateManager, method []byte, args ...[]byte) ([]byte, error) {
 	return sc.governanceStaking.ReadState(ctx, sm, method, args...)
 }
 
-func (sc *stakingCommittee) mergeDelegates(cand state.CandidateList, votes VoteTally) error {
+func (sc *stakingCommittee) mergeDelegates(list state.CandidateList, votes VoteTally) state.CandidateList {
 	// as of now, native staking does not have register contract, only voting/staking contract
 	// it is assumed that all votes done on native staking target for delegates registered on Ethereum
 	// votes cast to all outside address will not be counted and simply ignored
-	for i, c := range cand {
-		name := to12Bytes(c.CanName)
+	var merged state.CandidateList
```

Comment:
i think it's not necessary to clone the list, we just modify the Votes
but it's OK if you feel strongly to do so

Reply:
This is a coding practice problem. I believe it is a good way, and I would suggest doing similar thing for future code.